### PR TITLE
Feature/kas 4192 toevoegen secretaris aan pdf

### PR DESCRIPTION
**Pull request by StuyckRuben**
https://kanselarij.atlassian.net/browse/KAS-4192

- [x] first merge https://github.com/kanselarij-vlaanderen/frontend-kaleidos/pull/1847

### CODE REVIEW

**Review by sergiofenoll — APPROVED** [review]
A different PR already has comments about the underlying changes w.r.t. the secretary dropdown. But the changes here (re-generating the report when the secretary changes) are good :+1:.

File: app/components/meeting/edit-meeting-modal.js [hunk 1]
```diff
@@ -211,6 +212,23 @@ export default class MeetingEditMeetingComponent extends Component {
...
   }
 
+  exportPdf = task(async (report) => {
+    const resp = await fetch(`/generate-decision-report/${report.id}`);
+    if (!resp.ok) {
+      this.toaster.error(this.intl.t('error-while-exporting-pdf'));
+      return;
+    }
+    return await resp.json();
+  });
```
Comment (ValenberghsSven) [inline app/components/meeting/edit-meeting-modal.js]: make a service for this task. (used in 2 places)

File: app/components/meeting/edit-meeting-modal.js [hunk 2]
```diff
@@ -211,6 +212,23 @@ export default class MeetingEditMeetingComponent extends Component {
...
+
+  async replaceReportFile(report, fileId) {
+    await deleteFile(report.file);
+    const file = await this.store.findRecord('file', fileId);
+    report.file = file;
+    report.modified = new Date();
+    await report.save();
+  }
+
```
Comment (ValenberghsSven) [inline app/components/meeting/edit-meeting-modal.js]: can be done in service

File: app/components/meeting/edit-meeting-modal.js [hunk 3]
```diff
@@ -211,6 +212,23 @@ export default class MeetingEditMeetingComponent extends Component {
...
   @dropTask
```
Comment (ValenberghsSven) [inline app/components/meeting/edit-meeting-modal.js]: normal task instead of dropTask

File: app/components/meeting/edit-meeting-modal.js [hunk 4]
```diff
@@ -230,7 +248,15 @@ export default class MeetingEditMeetingComponent extends Component {
...
-        yield decisionActivity.save();
+        yield decisionActivity.save(); 
+        const report = yield this.store.queryOne('report', {
+          'filter[:has-no:next-piece]': true,
+          'filter[decision-activity][:id:]': decisionActivity.id,
+        });
+        if (report) {
+          const fileMeta = yield this.exportPdf.perform(report);
+          yield this.replaceReportFile(report, fileMeta.id);
+        }
```
Comment (ValenberghsSven) [inline app/components/meeting/edit-meeting-modal.js]: service call `replaceGeneratedReport`

File: app/controllers/agenda/agendaitems/agendaitem/index.js [hunk 5]
```diff
@@ -96,10 +98,35 @@ export default class IndexAgendaitemAgendaitemsAgendaController extends Controll
...
+  });
+
+  async replaceReportFile(report, fileId) {
+    await deleteFile(report.file);
+    const file = await this.store.findRecord('file', fileId);
+    report.file = file;
+    report.modified = new Date();
+    await report.save();
+  }
+
```
Comment (ValenberghsSven) [inline app/controllers/agenda/agendaitems/agendaitem/index.js]: duplicate, use service call

**Review by ValenberghsSven — APPROVED** [review]
biased review

**Pull request merged**